test/e2e/integreatly_test.go

```diff
@@ -0,0 +1,141 @@
+package e2e
+
+import (
+	goctx "context"
+	"fmt"
+	//"sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+	"time"
+
+	"github.com/integr8ly/integreatly-operator/pkg/apis"
+	operator "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	retryInterval                    = time.Second * 5
+	timeout                          = time.Second * 60
+	deploymentRetryInterval          = time.Second * 30
+	deploymentTimeout                = time.Minute * 20
+	cleanupRetryInterval             = time.Second * 1
+	cleanupTimeout                   = time.Second * 5
+	installationCleanupRetryInterval = time.Second * 20
+	installationCleanupTimeout       = time.Minute * 2 //Longer timeout required to allow for finalizers to execute
+	intlyNamespacePrefix             = "intly-"
+)
+
+func TestIntegreatly(t *testing.T) {
+
+	logf.SetLogger(logf.ZapLogger(true))
+
+	installationList := &operator.InstallationList{}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, installationList)
+	if err != nil {
+		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+	}
+	// run subtests
+	t.Run("integreatly", func(t *testing.T) {
+		t.Run("Cluster", IntegreatlyCluster)
+	})
+
+}
+
+func waitForProductDeployment(t *testing.T, f *framework.Framework, ctx *framework.TestCtx, product, deploymentName string) error {
+	namespace := intlyNamespacePrefix + product
+	t.Logf("Checking %s:%s", namespace, deploymentName)
+
+	start := time.Now()
+	err := e2eutil.WaitForDeployment(t, f.KubeClient, namespace, deploymentName, 1, deploymentRetryInterval, deploymentTimeout)
+	if err != nil {
+		return err
```

Comment:
I am just curious what the message will look like here. I mean if it won't be too general.

Reply:
Will output something like:

```
integreatly_test.go:143: timed out waiting for the condition   
```
Which you can figure out in context:

```
....
wait_util.go:55: Waiting for availability of tutorial-web-app-operator deployment
wait_util.go:55: Waiting for availability of tutorial-web-app-operator deployment   
integreatly_test.go:143: timed out waiting for the condition 
```

But we can also wrap the error to make it clearer, Will do that.

Reply:
OK thanks. I think it could be sufficient but maybe worth checking in some next similar places.